pageserver/src/tenant/timeline.rs

```diff
@@ -4155,6 +4161,171 @@ impl Timeline {
         false
     }
 
+    /// Create image layers for Postgres data. Assumes the caller passes a partition that is not too large,
+    /// so that at most one image layer will be produced from this function.
+    async fn create_image_layers_for_rel_blocks(
+        self: &Arc<Self>,
+        partition: &KeySpace,
+        mut image_layer_writer: ImageLayerWriter,
+        lsn: Lsn,
+        ctx: &RequestContext,
+        img_range: Range<Key>,
+        start: &mut Key,
```

Comment:
Can we use a return value instead? The function can return something like `Result<ImageLayerCreationOutcome>` where `ImageLayerCreationOutcome` is:
```
struct ImageLayerCompactionOutcome {
  image: Option<ResidentLayer>,
  compacted_up_to: Key
};
```

And then you can handle the `start` update in the same function as it was declared.

Reply:
renamed compacted_up_to -> next_start_key because create image layer is not actually a compaction.